File: works/tmp/17386.cs
using static System.Console;
using System.Text;
using System.Collections;
using System.IO;
using System;

namespace Project1
{
    public class Class1
    {
        static int CCW(long x1, long y1, long x2, long y2, long x3, long y3)
        {
            long a = x1 * y2 + x2 * y3 + x3 * y1;
            long b = x2 * y1 + x3 * y2 + x1 * y3;

            if (a - b > 0) return 1;
            else return -1;
        }

        public static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(Console.OpenStandardInput());
            string[] _s = sr.ReadLine().Split();
            long x1 = long.Parse(_s[0]);
            long y1 = long.Parse(_s[1]);           
            long x2 = long.Parse(_s[2]);
            long y2 = long.Parse(_s[3]);
            _s = sr.ReadLine().Split();
            long x3 = long.Parse(_s[0]);
            long y3 = long.Parse(_s[1]);
            long x4 = long.Parse(_s[2]);
            long y4 = long.Parse(_s[3]);

            int abc = CCW(x1, y1, x2, y2, x3, y3);
            int abd = CCW(x1, y1, x2, y2, x4, y4);

            int dca = CCW(x4, y4, x3, y3, x1, y1);
            int dcb = CCW(x4, y4, x3, y3, x2, y2);

            if (abc * abd < 0 && dca * dcb < 0) Write(1);
            else Write(0);
        }
    }
}

File: works/tmp/17386.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun CCW(x1:Long, y1:Long, x2:Long, y2:Long, x3:Long, y3:Long): Int
{
    val a = x1*y2 + x2*y3 + x3*y1
    val b = x2*y1 + x3*y2 + x1*y3
    if(a-b > 0) return 1
    return -1
}

fun main()
{
    val sc = Scanner(System.`in`)
    val x1 = sc.nextLong(); var y1 = sc.nextLong()
    val x2 = sc.nextLong(); var y2 = sc.nextLong()
    val x3 = sc.nextLong(); var y3 = sc.nextLong()
    val x4 = sc.nextLong(); var y4 = sc.nextLong()

    val abc = CCW(x1, y1, x2, y2, x3, y3)
    val abd = CCW(x1, y1, x2, y2, x4, y4)

    val dca = CCW(x4, y4, x3, y3, x1, y1)
    val dcb = CCW(x4, y4, x3, y3, x2, y2)

    if(abc*abd < 0 && dca*dcb < 0) print(1)
    else print(0)
}